security/oc_cred.c

```diff
@@ -1249,9 +1307,17 @@ oc_sec_decode_cred(oc_rep_t *rep, oc_sec_cred_t **owner, bool from_storage,
                 (*owner)->owner_cred = true;
               }
               if (oc_apply_cred_cb) {
-                oc_apply_cred_cb(cr, on_apply_cred_data);
+                oc_sec_on_apply_cred_data_t cred_data = {
+                  cr,
+                  add_cred_data.replaced_cred,
+                  add_cred_data.created,
+                };
+                oc_apply_cred_cb(cred_data, on_apply_cred_data);
               }
             }
+            if (add_cred_data.replaced_cred) {
+              oc_sec_free_cred(add_cred_data.replaced_cred);
```

Comment:
maybe we need to free it at line 1294:
```c
  if (!oscore_ctx) {
   return false;
  }
```

Reply:
Good catch, I recall adding it there, but somehow it disappeared. Fixed.